src/styles/block/main.scss

```diff
@@ -0,0 +1,58 @@
+.page {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  max-width: $cellSize * 7 + 10;
+  margin-left: 4px;
+  padding-inline: 10px;
+  gap: $cellValue;
+
+  &__day {
+    display: flex;
+    box-sizing: border-box;
+    justify-content: center;
+    align-items: center;
+    width: $cellSize;
+    height: $cellSize;
+
+    background-color: #eee;
+    border: 1px solid black;
+    transition-duration: 0.5s;
+    cursor: pointer;
+
+    &:hover {
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+    }
+  }
+}
+
+// add day number
+@for $dayNumber from 1 through 31 {
+  .calendar__day:nth-child(#{$dayNumber})::before {
+    content: '#{$dayNumber}';
+  }
+}
+
+// delete over days
+.calendar--month-length-28 .calendar__day:nth-child(n + 29),
+.calendar--month-length-29 .calendar__day:nth-child(n + 30),
+.calendar--month-length-30 .calendar__day:nth-child(n + 31) {
+  display: none;
+}
+
+// position first day
+@each $name, $value in $daysOfWeek {
+  .calendar--start-day-#{$name} .calendar__day:first-child {
+    margin-left: ($value * ($cellSize + $cellValue)) - ($cellSize + $cellValue);
```

Comment:
The calculation for 'margin-left' seems overly complex. Consider simplifying the expression to improve readability and maintainability.